packages/core/src/utils.ts

```diff
@@ -390,6 +390,27 @@ function isZodType(value: unknown): value is z.ZodType {
     typeof (value as any).safeParse === 'function'
   );
 }
+/**
+ * Ensures a tool has an ID by generating one if not present
+ * @param tool - The tool to ensure has an ID
+ * @returns The tool with an ID
+ */
+export function ensureToolId(tool: ToolToConvert): ToolToConvert {
+  if (isVercelTool(tool) && !('id' in tool)) {
+    // Generate an ID based on the tool description or a random string if no description
+    const id = tool.description
+      ? tool.description.toLowerCase().replace(/\s+/g, '-')
```

Comment:
these descriptions can be really long sometimes so this might cause errors in some schemas. Maybe we can use a deterministic id hash fn to turn the description into an id instead? `someuuidhasher(tool.description)`

Reply:
yeah i'll look into that

Reply:
Added a hash function